lstchain/reco/utils.py

```diff
@@ -639,36 +639,43 @@ def add_delta_t_key(events):
 def get_effective_time(events):
     """
     Calculate the effective observation time of a set of real data events
-    from a sky observation. delta_t must be the time elapsed from the previous
-    *triggered* event, regardless of whether the list of events contains all
-    triggered events or not. It can be a list only of events which e.g. have
-    valid image parameters. Besides delta_t, each event must have
-    dragon_time, a timestamp (all in seconds)
+    from a sky observation. delta_t (s) must be the time elapsed from the
+    previous *triggered* event, regardless of whether the list of events
+    contains all triggered events or not. It can be a list only of events
+    which e.g. have valid image parameters. Besides delta_t, each event must
+    have dragon_time, a timestamp (s)
 
     Parameters
     ----------
-    events: pandas DataFrame
+    events: pandas DataFrame or astropy.table.QTable
+    If a dataframe, units are assumed to be seconds
 
     Returns
     -------
     t_eff: float
     t_elapsed: float
     """
+    timestamp = np.array(events['dragon_time'])
+    delta_t = np.array(events['delta_t'])
+
+    print('HOLA', type(timestamp))
+
+    if type(timestamp) is not u.Quantity:
```

Comment:
this works, but the canonical way of testing this would be `if not isinstance(timestamp, u.Quantity)`.

Reply:
However, you can just do:

```
timestamp = u.Quantity(timestamp, u.s, copy=False)
```
This will convert the series / array without units to having u.s but will also work when timestmap is already a quantity with time units.